net/nimble/controller/src/ble_ll_scan.c

```diff
@@ -151,13 +151,28 @@ static int
 ble_ll_aux_scan_cb(struct ble_ll_sched_item *sch)
 {
     struct ble_ll_scan_sm *scansm = &g_ble_ll_scan_sm;
+    uint8_t lls = ble_ll_state_get();
 
-    /* In case scan has been disabled just drop the scheduled item*/
-    if (!scansm->scan_enabled) {
+    /* In case scan has been disabled or there is other aux ptr in progress
+     * just drop the scheduled item
+     */
+    if (!scansm->scan_enabled || scansm->cur_aux_data) {
         ble_ll_scan_aux_data_free(sch->cb_arg);
         goto done;
     }
 
+    /* Check if there is no aux connect sent. If so drop the sched item */
+    if (lls == BLE_LL_STATE_INITIATING && ble_ll_conn_init_aux_conn_is_sent()) {
+        ble_ll_scan_aux_data_free(sch->cb_arg);
+        goto done;
+    }
+
+    if (lls != BLE_LL_STATE_STANDBY) {
+        ble_phy_disable();
+        ble_ll_wfr_disable();
+        ble_ll_state_set(BLE_LL_STATE_STANDBY);
```

Comment:
no because we are here only in 3 states:
* BLE_LL_STATE_STANDBY
* BLE_LL_STATE_INITIATING
* BLE_LL_STATE_SCANNING

I will put comment about that.